### PR TITLE
Add mpi4py sub-communicator example

### PR DESCRIPTION
**Pull request by alsrgv**

### CODE REVIEW

**Comment by alsrgv** [comment]
cc @IgorWilbert

**Comment by abditag2** [comment]
@alsrgv curious, why would this be ever needed?

**Comment by alsrgv** [comment]
@abditag2, in HPC environment users often run their whole optimization problem as a single MPI application.  The optimization problem may be shardable into sub-problems, each of which could run its own Horovod training.

File: README.rst [hunk 1]
```diff
@@ -252,6 +252,23 @@ You can check for MPI multi-threading support by querying the ``hvd.mpi_threads_
     from mpi4py import MPI
     assert hvd.size() == MPI.COMM_WORLD.Get_size()
 
+Horovod can also be initialized with an `mpi4py` sub-communicator, in which case each sub-communicator
```
Comment (sblotner) [inline README.rst]: You can also initialize Horovod with ...

File: docs/summary.rst [hunk 2]
```diff
@@ -284,6 +284,23 @@ You can check for MPI multi-threading support by querying the ``hvd.mpi_threads_
     from mpi4py import MPI
     assert hvd.size() == MPI.COMM_WORLD.Get_size()
 
+Horovod can also be initialized with an `mpi4py` sub-communicator, in which case each sub-communicator
```
Comment (sblotner) [inline docs/summary.rst]: Same comment as above

**Pull request merged**